File: src/test/java/net/slipp/web/UserFormTest.java
package net.slipp.web;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

import org.junit.Test;

public class UserFormTest {
    @Test
    public void valid_userId() {
        assertUserId("ni", true);
        assertUserId("테스트", true);
        assertUserId("박.재.성", true);
        assertUserId("나만의 닉네임을 가지고", true);
        assertUserId(null, false);
        assertUserId("", false);
        assertUserId("n", false);
        assertUserId("나만의 닉네임을 가지고 테", false);
    }
    
    private void assertUserId(String userId, boolean expected) {
        String email = "email@gmail.com";
        
        UserForm form = new UserForm(userId, email);
        assertThat(form.isValid(), is(expected));
    }
}

File: src/test/scala/net/slipp/web/UserFormTest.scala
package net.slipp.web

import org.junit.Assert._
import org.hamcrest.CoreMatchers._
import org.junit.Test

class UserFormTest {
  @Test def create() {
    val userForm = new UserForm
    userForm.setUserId("userId")
    assertThat(userForm.getUserId(), is("userId"))
  }

  @Test def valid_userId() {
    assertUserId("ni", true)
    assertUserId("테스트", true)
    assertUserId("박.재.성", true)
    assertUserId("나만의 닉네임을 가지고", true)
    assertUserId(null, false)
    assertUserId("", false)
    assertUserId("n", false)
    assertUserId("나만의 닉네임을 가지고 테", false)
  }

  def assertUserId(userId: String, expected: Boolean) {
    val email = "email@gmail.com"
    val form = new UserForm(userId, email)
    assertThat(form.isValid(), is(expected))
  }
}